File: shopy-api/Models/Withdrawal.cs
namespace shopy_api.Models;

public class Withdrawal : ISoftDeletable
{
    public Guid Id { get; set; }

    public Guid StoreId { get; set; }
    public Store Store { get; set; } = null!;

    public Guid BankAccountId { get; set; }
    public BankAccount BankAccount { get; set; } = null!;

    public decimal Amount { get; set; }
    public decimal AdminFee { get; set; }
    public decimal NetAmount { get; set; }

    public WithdrawalStatus Status { get; set; } = WithdrawalStatus.Pending;
    public string? RejectReason { get; set; }

    public DateTime RequestedAt { get; set; }
    public DateTime? ProcessedAt { get; set; }

    public bool IsDeleted { get; set; }
}

File: shopy-seller/lib/models/finance/withdrawal.dart
class Withdrawal {
  final String id;
  final int amount;
  final int adminFee;
  final int netAmount;
  final String status;
  final String bankName;
  final String accountNumberMasked;
  final DateTime requestedAt;
  final DateTime? processedAt;

  const Withdrawal({
    required this.id,
    required this.amount,
    required this.adminFee,
    required this.netAmount,
    required this.status,
    required this.bankName,
    required this.accountNumberMasked,
    required this.requestedAt,
    this.processedAt,
  });

  factory Withdrawal.fromJson(Map<String, dynamic> json) {
    return Withdrawal(
      id: json['id'] as String,
      amount: (json['amount'] as num).round(),
      adminFee: (json['adminFee'] as num).round(),
      netAmount: (json['netAmount'] as num).round(),
      status: json['status'] as String,
      bankName: json['bankName'] as String,
      accountNumberMasked: json['accountNumberMasked'] as String,
      requestedAt: DateTime.parse(json['requestedAt'] as String),
      processedAt: json['processedAt'] == null ? null : DateTime.parse(json['processedAt'] as String),
    );
  }
}
